### PR TITLE
enhance: change opendal as compile configurable

### PR DESCRIPTION
**Pull request by zhagnlu**
#30373

### CODE REVIEW

File: internal/core/thirdparty/CMakeLists.txt [hunk 1]
```diff
@@ -36,7 +36,10 @@ add_subdirectory(boost_ext)
 add_subdirectory(rocksdb)
 add_subdirectory(rdkafka)
 add_subdirectory(simdjson)
-add_subdirectory(opendal)
+if (USE_OPENDAL)
+    message(STATUS "xxxx")
```
Comment (czs007) [inline internal/core/thirdparty/CMakeLists.txt:40]: remove this ? or refine the message

Comment (zhagnlu) [reply internal/core/thirdparty/CMakeLists.txt:40]: OK, remove it

**Comment by zhagnlu** [comment]
rerun ut

**Pull request merged**